runtime/src/main/cpp/TestSupport.hpp

```diff
@@ -15,4 +21,46 @@ constexpr int kDefaultThreadCount = 10;
 constexpr int kDefaultThreadCount = 100;
 #endif
 
+// Performs minimal initialization of the current thread memory subsystem.
+// This initilization is enough for running tests for the C++ part for the stdlib.
+//  - For the new MM: registeres the current thread in the thread registry.
+//  - For the legacy MM: does nothing, returns nullptr.
+MemoryState* InitMemoryForTests();
+
+// Deinitiliazes memory subsystem used for C++ stdlib tests.
+//  - For the new MM: unregisteres the current thread, nullify current thread data.
+//  - For the legacy MM: does nothing.
+void DeinitMemoryForTests(MemoryState* state);
+
+// Scopely initializes the runtime for tests.
+class ScopedRuntimeInit : private kotlin::Pinned {
+public:
+    ScopedRuntimeInit() : memoryState_(InitMemoryForTests()) {
+        test_support::InitRuntimeForTests(memoryState());
+    }
+    ~ScopedRuntimeInit() {
+        test_support::DeinitRuntimeForTests();
+        DeinitMemoryForTests(memoryState_);
+    }
+
+    MemoryState* memoryState() { return memoryState_; }
+private:
+    MemoryState* memoryState_;
+};
+
+// Runs the given function in a separate thread with minimally initialized runtime.
+inline void RunInNewThread(std::function<void(MemoryState*)> f) {
```

Comment:
Do we need `std::function` here instead of a usual template? If it's for the overload below, we can just ignore the argument at each use site.

Reply:
It's rather for [this](https://github.com/JetBrains/kotlin-native/blob/5778e42fec07ef2fd37856203febe94e0a89eff0/runtime/src/mm/cpp/TestSupport.hpp#L13) overload because new MM tests usually work with `ThreadData`, not `MemoryState`. Of course we could convert `MemoryState` to `ThreadData` at use sites but it looks like an unneeded boilerplate. 
I haven't find a way to keep both this overload and usage of a usual template.

Reply:
I thought about it but decided that having functions with the same name and almost the same behaviour in different namespaces will be less clear and less convenient.